R/DBConnection.R

```diff
@@ -106,6 +106,97 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' Sentinel value to indicate the early termination of
+#' \code{\link{dbGetChunkedQuery}}
+#' @export
+dbBreak <- structure(list(), class = "dbi_break_sentinel")
+
+dbFetchChunked <- function(rs, n, callback) {
+  while (TRUE) {
+    chunk <- dbFetch(rs, n = n)
+    if (nrow(chunk) == 0) {
+      return(chunk)
+    }
+    result <- callback(chunk)
```

Comment:
Personally, I prefer `callback(chunk)` to `callback(chunk = chunk)` because I like giving people flexibility to call the "main" parameter whatever they like--in some cases their code might be more readable if they can say `function(transactions) {` instead of `function(chunk) {`.. However I do think it's a good idea to call with the name when talking about other (minor) arguments.